### DISCUSSION TITLE
Share Your Experience with Repomix

### DISCUSSION THREAD

**Discussion by yamadashy** [yamadashy/repomix#154] [General]
Hey everyone! We'd love to hear how you're using Repomix with AI tools.
Your experiences could inspire others and help improve Repomix!

Feel free to share:
- Which AI tool you used (Claude, ChatGPT, Gemini, etc.)
- The prompt you found effective
- How it helped you
- Any tips or tricks you discovered

Share in whatever format works for you - keep it casual!

**Comment by yamadashy** [comment]
I've been using Repomix extensively for all aspects of Repomix's own development.

I primarily use Claude (Web version), creating a dedicated Project with the Repomix output file for continuous consultation. When Claude hits usage limits, I switch to Gemini as a backup.

Here's how I use it in my daily development:

- Implementation Support
  - Discussing implementation approaches
  - Getting code suggestions - works well for most tasks unless they're complex. Sometimes the generated code misses some functionality due to my incomplete requirements, but it's easy to explain and get the missing parts added
- Test Generation
  - I explain what requirements and cases need to be tested, and about 60-80% of the generated tests work right away. For complex scenarios requiring extensive mocking or external dependencies, the generated tests often need manual adjustments. I investigate what didn't work and iteratively refine the tests with the AI's help
- Library Understanding
  - Pack unfamiliar libraries to understand their usage. Super helpful with limited documentation
- Git & Documentation
  - Reviewing diffs to write better commit messages, PR descriptions, and GitHub release notes

Some tips that worked well for me:
- Keep your code modular and small - makes it easier for AI to understand and generate quality code
- For complex features, start with design discussions first, then move to implementation

From my experience with code generation capabilities:
OpenAI o1 > Claude 3.5 Sonnet > Gemini 1.5 Pro > GPT-4o

**Reply by massdo** [reply]
Did you try Cursor? On my side, I use it, and it feels like the quality of the code is lower compared to directly using Claude. I wonder if the lower quality is intentional to save costs, but I’m curious if anyone else feels the same.

**Comment by kevinlearnscoding** [comment]
I just learned about Repomix about an hour ago. 
While I've had some trouble setting it up (I can't seem to properly set the output path or file name) I have managed to get it to run on a remote repo and produce a Markdown file in .txt format. 
I've then passed that file onto ChatGPT to extract the parts of the code that I need, and it's helping me build my first program. 


I'm just learning code - (like this is day 4 or 5 or me learning code) so having language model AI tools available to ask questions to, has been absolutely invaluable. The fact that they can also review code, and write code... truly incredible.

**Reply by WSINTRA** [reply]
Be careful, using LLM's to 'learn to code' makes you very lazy on technical details and you will suffer, just like having a phone save contacts leads to poor memory for phone numbers, having an LLM generate code leads to poor ability to write code independent of the LLM. 
Be sure to augment your learning with books on whatever language you are learning. Maybe try building a small TODO app with LLM and without. Get creative but remember it will be a hindrance if you don't address it.

**Comment by robcsaszar** [comment]
I've been using it on-ff for about one week now, with Claude (I didn't know you could pass an output file to ChatGPT, I'll give that a try).
To how I'm using it:
- in `package.json` I have these two scripts defined:
```json
"repomix": "node repomix/repomix.cjs",
"repomix:all": "node repomix/repomix.cjs --all",
```
- in my `repomix.cjs` file I have the following logic so I can create timestamped output files:
```javascript
const dateFns = require('date-fns');
const exec = require('node:child_process').execSync;

const date = dateFns.format(new Date(), 'yyyy-MM-dd');
const repomix = `pnpm dlx repomix --config repomix/repomix.config.json --include "src" --output "repomix/output/${date}-output.md"`;
const repomixAll = `pnpm dlx repomix --config repomix/repomix.config.json --include "**/*" --output "repomix/output/${date}-output-all.md"`;
const command = process.argv.includes('--all') ? repomixAll : repomix;

exec(command, { stdio: 'inherit' });
```
- I added an `instructions.md` file for my "special" instructions, which is basically some coding guidelines I follow
- finally, the `repomix.config.json` file is pretty basic as per your documentation:
```json
{
  "output": {
    "filePath": "repomix/output/output.md",
    "style": "markdown",
    "removeComments": false,
    "removeEmptyLines": false,
    "topFilesLength": 5,
    "showLineNumbers": false,
    "instructionFilePath": "repomix/instructions.md"
  },
  "include": [],
  "ignore": {
    "useGitignore": true,
    "useDefaultPatterns": true,
    "customPatterns": [
      "**/*.i18n.*",
      "**/*.constants.*",
      "**/*.config.*",
      "**/*.env.*",
      "**/*.secret.*",
      "**/*.svg"
    ]
  },
  "security": {
    "enableSecurityCheck": true
  }
}
```

I used it to implement a SolidJS data table in my Astro project with sorting, caching, table state, and filtering. It helped tons, although my current (non-all) output takes about 51% of memory, and Claude seems to have a mind of its own and seems to ignore my project structure these past 2–3 days, plus, it's not up-to-date on Astro so that's a hit & miss sometimes, and it takes them aaaages to respond on a new chat. Haven't tried tests yet, but I intend to do that. Do you have any tips for an approach there?

Also, since you use it extensively with Claude as well, are you using [ClaudeSync](https://github.com/jahwag/ClaudeSync) or just upload the file manually, like I do?

I think I should set up an output for a particular set of files, so I can focus on a specific part of my app so the responses are more targeted, but I haven't looked into that yet, I suppose I can do that with the --include.

**Reply by Pipboyguy** [reply]
+1 for claudesync

**Reply by danja** [reply]
claudesync? wtf!! Christmas really has come early. 
Thank you.

**Comment by rtjohn** [comment]
Can anyone give me a sense of what repo size you're finding success with?  For reference, the repomix-output.txt file for the project I want to use is 1.2MB in size. I keep running into token limits for most models with the OpenAI API.

**Comment by danja** [comment]
I've been using s/repopack/**repomix** *extensively* with Claude Projects. I've found it incredibly useful, many thanks @yamadashy, excellent work! (if you would like praise expressed more formally/for promotional materials, let me know). 

The main project 'Transmissions' I'm working on at the moment has LOC & repomix stats as below. I'm predominantly using node, occasional bits of browser stuff (also a toy language 'sheltopusik' I have in-progress, the less said about that the better - hence the instantly forgettable name). 
Transmissions is still lacking key features and is neither nor stable nor documented enough for anyone else to have a use for it. But there is some intersection with repomix, in that I'm trying to find a sweetspot for working with LLMs. It's a pipeliney thing with major decoupling, the intention being that any part that needs work on can be comfortably understood in the current size of context windows.  This also means I've got a ton of other related projects, each with their own repo, varying sizes. 
 
The repomix config for transmissions isn't far off defaults regarding code, except I do have several additions to the ignore section - like a `docs` dir that currently includes masses of irrelevant (and redundant) styling stuff.  But the (hopefully) useful stuff takes up 55% of the Claude Project knowledge (funny, it was 63% but just now I noticed there was another irrelevant dir - the Top 5 is a great feature!). In addition to this I have a smaller associated repo that notches it up to 73%. This, with a fairly verbose system prompt, seems to push the limits of utility. So trying to connect another project, which may also produce a sizable repomix, with Transmissions was a problem. 

My dev workflow for a *New Feature* (more or less)  is figuring out a prompt describing my requirements for it, repomixing (good verb) and replacing the project's current knowledge, starting a new chat session in the project. This I follow for maybe a dozen interactions, 6 or so artifacts, then halt, however prematurely, and ask for a handover document (I have a bunch of commands in the system prompt, `ho` gives me a summary as an artifact, including metadata in RDF/Turtle). Even being careful, with the *Big Knowledge*, I often hit the token limit. *Claudio gets another walk.*  
Something I've found useful when working with the *other* project as the main context, but still wishing Claude to have knowledge of Transmissions for integration purposes, is to have this in my `package.json` : 

```json
{
...
  "scripts": {
    "rp": "repomix -c repomix.config-small.json . && repomix -c repomix.config-large.json . ",
   ...
   }
}
```
`repomix.config-small.json` has a lot more in the `ignore` section, stripping things back to a bare minimum, used in auxiliary projects. `repomix.config-large.json` has everything relevant, used in 'transmissions' itself. 

(I'm leaving it as legacy `rp`, I'm all-Linux here `rm` is rarely wise :)

fyi @yamadashy, I plan to use chunks of repomix, pulled apart into small pieces, in my Transmissions pipeliney thing. I've not actually looked yet how things are exposed, but ideally I'd like to use it as a lib, so one node on my processing pipeline loads the config, another build the path filters, aother does the treewalking etc etc. 

Whatever, thanks again, keep up the good work!
---


```sh
  Pack Summary:
────────────────
  Total Files: 264
  Total Chars: 314527
 Total Tokens: 73062
       Output: ./repopack-transmissions-large.txt
     Security: ✔ No suspicious files detected

transmissions$ cloc src
     231 text files.
     166 unique files.                                          
      65 files ignored.

github.com/AlDanial/cloc v 1.98  T=3.00 s (55.4 files/s, 2928.8 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
JavaScript                     115           1035           1082           3971
Markdown                        28            461              5           1113
HTML                             2             17              0            763
JSON                            14              0              0            292
Bourne Shell                     1              8              9              9
Text                             6              0              0              7
-------------------------------------------------------------------------------
SUM:                           166           1521           1096           6155
-------------------------------------------------------------------------------
```

**Comment by BestofthebestinAI** [comment]
Instead of showing my project, I want to inquiy and ask for your advice for the following:
How best to feed AI your whole code as a base code to generate more code depending on it?
For example:
- Let's say someone wrote some interesting NEW library, with its own functions and modules.
- I want AI to learn it (the library is new AI does not know it yet)
- How best to use repomix to FEED that into the AI? I am assuming xml is not a good way? I am totally new did not use repomix when writing these liens yet, but interested in hearing your experiences or advice?

**Comment by trydhrumil-cmyk** [comment]
I just heard about this tool which helps to generate the markdown file for my existing repo. i just tried with some small project and found its generating md file which contain full code base. i still dont understand do we need to feed the existing whole code base as system prompt( as i am going to put this file as Agents.md in my repo) to my agents to generate new code. i dont think so we need to generate the full code to tell the agents how to generate the new code. 

i am looking for somethign which will read my existing code and tell the meaning full inforamation to AI agents about what to look for in my existing code not full code and understand it. 

hope the next enhancement will give such information.

**Reply by jerell2isekai** [reply]
for me, I use repomix to packed and compressed codebase into a single markdown file, and use it as a additional memory file, only ask AI (claude、codex、gemini cli) to read it when I need it to do. and it works very well.

**Comment by homelab-00** [comment]
Just wanted to say hi and share my excitement for this tool, it makes life so much easier. No longer have to duplicate text files not ending in one of the popular file extensions and having to append .txt to them so the stupid Google AI Studio interface will accept them.

Thanks!
